crates/sqlexec/src/planner/physical_plan/client_send.rs

```diff
@@ -0,0 +1,274 @@
+use crate::errors::Result;
+use crate::remote::client::RemoteSessionClient;
+use datafusion::arrow::array::UInt64Array;
+use datafusion::arrow::datatypes::{DataType, Field, Schema};
+use datafusion::arrow::ipc::writer::FileWriter as IpcFileWriter;
+use datafusion::arrow::record_batch::RecordBatch;
+use datafusion::error::{DataFusionError, Result as DataFusionResult};
+use datafusion::execution::TaskContext;
+use datafusion::physical_expr::PhysicalSortExpr;
+use datafusion::physical_plan::{
+    stream::RecordBatchStreamAdapter, DisplayAs, DisplayFormatType, ExecutionPlan, Partitioning,
+    SendableRecordBatchStream, Statistics,
+};
+use futures::{Stream, StreamExt};
+use parking_lot::Mutex;
+use protogen::gen::rpcsrv::service;
+use std::any::Any;
+use std::fmt;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use uuid::Uuid;
+
+/// Execution plan for sending batches to a remote node.
+#[derive(Debug)]
+pub struct ClientExchangeInputSendExec {
+    broadcast_id: Uuid,
+    client: RemoteSessionClient, // TODO: Extension
+    input: Arc<dyn ExecutionPlan>,
+}
```

Comment:
Adding in an extension won't be necessary right now since these will always be executed on the client side (and so will always have the client available).